### PR TITLE
Functional tests - Fix and add some steps to ' Customer block & Product block' on create order page BO

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix and add some steps to ' Customer block & Product block' on create order page BO
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no need
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26295)

### CODE REVIEW

File: tests/UI/pages/BO/international/taxes/index.js [hunk 1]
```diff
@@ -318,6 +318,20 @@ class Taxes extends BOBasePage {
     return this.getAlertSuccessBlockParagraphContent(page);
   }
 
+  /**
+   * Enable/Disable eco tax
+   * @param page
+   * @param enableEcoTax
```
Comment (Progi1984) [inline tests/UI/pages/BO/international/taxes/index.js]:
```suggestion
   * @param page {Page} Browser tab
   * @param enableEcoTax {boolean}
```

File: tests/UI/pages/BO/orders/view.js [hunk 2]
```diff
@@ -595,6 +598,15 @@ class Order extends BOBasePage {
     return this.elementVisible(page, `${this.addProductAddButton},disabled`, 1000);
   }
 
+  /**
+   * Is add product table row visible
+   * @param page
```
Comment (Progi1984) [inline tests/UI/pages/BO/orders/view.js]:
```suggestion
   * @param page {Page} Browser tab
```

**Review by Progi1984 — COMMENTED** [review]
@nesrineabdmouleh A small feedback about some JSDocs but it's good

File: tests/UI/pages/BO/international/taxes/index.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/international/taxes/index.js]: Done thx :+1:

File: tests/UI/campaigns/data/faker/cartRule.js [hunk 3]
```diff
@@ -96,6 +100,17 @@ class CartRuleData {
...
+    /** @type {string} Object to apply discount on it */
+    this.applyDiscountTo = cartRuleToCreate.applyDiscountTo || 'None';
+
+    /** @type {string} Name of the product to apply cart rule */
+    this.product = undefined;
+
+    /** @type {string} Apply discount to which object */
+    if (this.applyDiscountTo === 'Specific product') {
+      this.product = cartRuleToCreate.product || faker.random.arrayElement(ProductsNames);
+    }
```
Comment (boubkerbribri) [inline tests/UI/campaigns/data/faker/cartRule.js]:
```suggestion
    /** @type {string|undefined} Name of the product to apply cart rule */
    this.product = undefined;
    if (this.applyDiscountTo === 'Specific product') {
      this.product = cartRuleToCreate.product || faker.random.arrayElement(ProductsNames);
    }
```

File: tests/UI/campaigns/data/faker/product.js [hunk 4]
```diff
@@ -62,6 +62,12 @@ class ProductData {
     /** @type {string} Tac rule to apply the product */
     this.taxRule = productToCreate.taxRule || 'FR Taux standard (20%)';
 
+    /** @type {string} EcoTax tax included of the product */
+    this.ecoTax = productToCreate.ecoTax === undefined ? faker.random.number({
+      min: 10,
+      max: 20,
+    }) : productToCreate.ecoTax;
```
Comment (boubkerbribri) [inline tests/UI/campaigns/data/faker/product.js]:
```suggestion
    /** @type {number} EcoTax tax included of the product */
    this.ecoTax = productToCreate.ecoTax === undefined
      ? faker.random.number({min: 10, max: 20})
      : productToCreate.ecoTax;
```

File: tests/UI/campaigns/data/faker/product.js [hunk 5]
```diff
@@ -75,10 +81,12 @@ class ProductData {
       : productToCreate.minimumQuantity;
 
     /** @type {string} Stock location of the product */
-    this.stockLocation = productToCreate.stockLocation || '';
+    this.stockLocation = productToCreate.stockLocation || 'stock 1';
 
     /** @type {string} Low stock level of the product */
```
Comment (boubkerbribri) [inline tests/UI/campaigns/data/faker/product.js]:
```suggestion
    /** @type {number} Low stock level of the product */
```

File: tests/UI/campaigns/functional/BO/02_orders/01_orders/viewAndEditOrder/01_customerBlock.js [hunk 6]
```diff
@@ -331,20 +318,94 @@ describe('BO - Orders - view and edit order : Check and edit customer block', as
...
+
+      const shippingAddress = await viewOrderPage.getInvoiceAddress(page);
+      await expect(shippingAddress)
+        .to.contain(firstAddressData.firstName)
+        .and.to.contain(firstAddressData.lastName)
+        .and.to.contain(firstAddressData.address)
+        .and.to.contain(firstAddressData.postalCode)
+        .and.to.contain(firstAddressData.city)
+        .and.to.contain(firstAddressData.country);
+    });
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/01_orders/viewAndEditOrder/01_customerBlock.js:364]: :+1:

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh & @boubkerbribri

**Pull request merged**